### DISCUSSION TITLE
Populate a database with datas from a graphql api

### DISCUSSION THREAD

**Discussion by PazzeG** [prisma/prisma#14116] [Q&A]
Hello, guys I'm looking for, as the topic name says, to populate a database with datas coming from an already exist graphql API [https://anilist.gitbook.io/anilist-apiv2-docs/overview/graphql/getting-started](url). 

My project is using Next.js & PostgreSQL and of course to link these two technologies Prisma. I managed to retrieve datas from the API and display these. Now I just want to store these datas in my Postgres database. So you know I'm using the client-side rendering and not the other methods (SSR, SSG)
All documentations I found was about creating an API then post to the database which is not my case, and also documentation about create & createMany queries but with hardcoded datas. My research doesn't help me to see clear so I come here to find anyone who is willing to guide me. If have a clue I'll appreciate. Thank you =) !

To illustrate here a piece of code : 
```
import { useQuery, gql } from "@apollo/client";

// query to retrieve 3 animes

const QUERY = gql`
  query GetFirstsThree{
    Page(page: 1, perPage: 3) {
      media {
        title {
          userPreferred
        }
      }
    }
  }
`;

export default function AnimList() {
  const { data, loading, error } = useQuery(QUERY);
  
  if (loading) {
    return <h2>Loading...</h2>;
  }
  if (error) {
    console.error(error);
    return null;
  }

  const medias = data.Page.media;

  return (
    <div className="mainGrid">
      {medias.map((value) => {
        return (
            <p>{value.title.userPreferred} </p>
        )
      })}
    </div>
    );
}
```

**Comment by ruheni** [comment]
Hey 👋🏽

> Now I just want to store these datas in my Postgres database

If I understand correctly, you wish to learn how you can write to a database using a GraphQL API (using Prisma under the hood). To write to your database, you need to define a [GraphQL mutation](https://www.howtographql.com/basics/2-core-concepts/). I would recommend going through either the [TypeScript-Apollo tutorial](https://www.howtographql.com/typescript-apollo/0-introduction/) or [Fullstack App With TypeScript, PostgreSQL, Next.js, Prisma & GraphQL: Data Modeling](https://www.prisma.io/blog/fullstack-nextjs-graphql-prisma-oklidw1rhw).

Let me know if you would need any further clarification. 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
